### ISSUE TITLE
MeshBrillouinZone thinks it is a cyclic_lattice

### ISSUE THREAD

**Issue by HugoStrand** [TRIQS/triqs#518]
Dear all,

The `MeshBrillouinZone` mesh claims to be a `cyclic_lattice` when asked.
```python
from pytriqs.gf import MeshBrillouinZone
from pytriqs.lattice import BrillouinZone, BravaisLattice
bz = BrillouinZone(BravaisLattice([[1,0],[0,1]]))
bzmesh = MeshBrillouinZone(bz, n_k=15)
print type(bzmesh)
print bzmesh
```
gives the output
```
<type 'pytriqs.gf.meshes.MeshBrillouinZone'>
cyclic_lattice of size (15 15 1)
```
Is this a bug in the `__str__` method of the wrapped bz mesh?

Best, Hugo

**Comment by tayral** [comment]
Hi Hugo,
No, it is because both gf_mesh<brillouin_zone> and gf_mesh<cyclic_lattice> derive (in c++) from an object "cluster_mesh" (for reasons which I can give you more details about), whose print method is "cyclic lattice of size ...". See https://github.com/TRIQS/triqs/blob/unstable/triqs/lattice/cluster_mesh.hpp
One could overload it for both classes to make it clearer.
Best
Thomas

**Comment by Wentzell** [comment]
Fixed by 523040ef
